packages/shorebird_cli/lib/src/command_pipeline/command_pipeline.dart

```diff
@@ -0,0 +1,18 @@
+import 'package:shorebird_cli/src/command_pipeline/command_pipelines.dart';
+
+class CommandPipeline {
+  CommandPipeline({
+    required this.steps,
+    PipelineContext? context,
```

Comment:
Do we ever want anyone to be able to inject a context? Is this just for testing?

Reply:
The idea is that this will be seeded by the release command, e.g., `final context = PipelineContext()..provide(() => results);`